### PR TITLE
Android 14 updates

### PR DESCRIPTION
**Pull request by ShujathMohd**

### CODE REVIEW

File: vintf/manifest.xml [hunk 1]
```diff
@@ -1,4 +1,5 @@
-<manifest version="1.0" type="device" target-level="3">
+<manifest version="1.0" type="device" target-level="4">
+   <kernel target-level="5" />
```
Comment (MarijnS95) [inline vintf/manifest.xml:2]:
```suggestion
    <kernel target-level="5" />
```

**Comment by MarijnS95** [comment]
Can you drop the leading spaces before `:` in commit titles? It's not standard.

File: vintf/manifest.xml
Comment (MarijnS95) [inline vintf/manifest.xml:1]: Have you checked that we are all set for FCM 4? Or just bumping this "because the build tool says so"?

Comment (ShujathMohd) [reply vintf/manifest.xml:1]: Nice catch, Let me verify 

https://android.googlesource.com/platform/hardware/interfaces/+/main/compatibility_matrices/compatibility_matrix.4.xml

Comment (ShujathMohd) [reply vintf/manifest.xml:1]: We are compatible with FCM 4 except android.hardware.thermal 1.0

**Comment by ShujathMohd** [comment]
Screenshot_20231025-190542
Thermals are showing as expected.

**Comment by MarijnS95** [comment]
> Can you drop the leading spaces before `:` in commit titles? It's not standard.

:eyes:

**Comment by ShujathMohd** [comment]
> frameworks/opt/net/wifi/libwifi_hal/wifi_hal_common.cpp:194:7: error: no matching function for call to 'wifi_change_driver_state' if (wifi_change_driver_state(WIFI_DRIVER_STATE_ON) < 0) { ^~~~~~~~~~~~~~~~~~~~~~~~

https://github.com/sonyxperiadev/device-sony-edo/pull/52

**Review by jerpelea — APPROVED** [review]

**Comment by jerpelea** [comment]
sorry. I missed them

**Comment by jerpelea** [comment]
@ShujathMohd did you fix extended settings and modemconfig?

**Comment by ShujathMohd** [comment]
> @ShujathMohd did you fix Extended Settings and modemconfig?

Nope, didnt get time to look into them

**Comment by MarijnS95** [comment]
> @ShujathMohd did you fix Extended Settings and modemconfig?

They build for me; is there a problem at runtime?

**Comment by jerpelea** [comment]
API of sysprop_library SomcModemProperties doesn't match with current.txt
Please update current.txt by:
m SomcModemProperties-dump-api && mkdir -p vendor/oss/opentelephony/ModemConfig/sysprop/api && rm -rf vendor/oss/opentelephony/ModemConfig/sysprop/api/SomcModemProperties-current.txt && cp -f out/soong/.intermediates/vendor/oss/opentelephony/ModemConfig/sysprop/SomcModemProperties_sysprop_library/api-dump.txt vendor/oss/opentelephony/ModemConfig/sysprop/api/SomcModemProperties-current.txt

**Comment by jerpelea** [comment]
@ShujathMohd 
packages/apps/ExtendedSettings/src/sonyxperiadev/extendedsettings/ExtendedSettingsFragment.java:73: error: cannot find symbol
    private static final long BUILT_IN_DISPLAY_ID_MAIN = SurfaceControl.getPhysicalDisplayIds()[0];
                                                                       ^
  symbol:   method getPhysicalDisplayIds()
  location: class SurfaceControl
packages/apps/ExtendedSettings/src/sonyxperiadev/extendedsettings/ExtendedSettingsFragment.java:281: error: cannot find symbol
        IBinder displayHandle = SurfaceControl.getPhysicalDisplayToken(dispId);
                                              ^
  symbol:   method getPhysicalDisplayToken(int)
  location: class SurfaceControl
packages/apps/ExtendedSettings/src/sonyxperiadev/extendedsettings/ExtendedSettingsFragment.java:282: error: incompatible types: IBinder cannot be converted to long
        SurfaceControl.DynamicDisplayInfo displayInfo = SurfaceControl.getDynamicDisplayInfo(displayHandle);
                                                                                             ^
packages/apps/ExtendedSettings/src/sonyxperiadev/extendedsettings/ExtendedSettingsFragment.java:413: error: cannot find symbol
        IBinder displayHandle = SurfaceControl.getPhysicalDisplayToken(BUILT_IN_DISPLAY_ID_MAIN);
                                              ^
  symbol:   method getPhysicalDisplayToken(long)
  location: class SurfaceControl
packages/apps/ExtendedSettings/src/sonyxperiadev/extendedsettings/ExtendedSettingsFragment.java:430: error: incompatible types: IBinder cannot be converted to long
        SurfaceControl.DynamicDisplayInfo displayInfo = SurfaceControl.getDynamicDisplayInfo(displayHandle);
                                                                                             ^
packages/apps/ExtendedSettings/src/sonyxperiadev/extendedsettings/ExtendedSettingsFragment.java:451: error: cannot find symbol
        SurfaceControl.setDisplaySize(displayHandle, width, height);
                      ^
  symbol:   method setDisplaySize(IBinder,int,int)

**Comment by MarijnS95** [comment]
@jerpelea the `SurfaceControl` portion of ES will need some love, we shouldn't just mess with it :)

**Comment by jerpelea** [comment]
@MarijnS95 let's remove it for now

**Comment by MarijnS95** [comment]
Sure, please remove it instead of having something that doesn't work. It needs to be revisited entirely to work with the modern KMS and SDE composer anyway.

**Pull request merged**